src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -9314,5 +9314,495 @@ public unsafe struct CAPROPVARIANT
+                // using unsafe X = int*[];
+                Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using unsafe X = int*[];").WithLocation(2, 1));
+        }
+
+        [Fact]
+        public void TestUnsafeAlias10()
+        {
+            var csharp = @"
+using X = int*[];
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics(
+                // (2,1): hidden CS8019: Unnecessary using directive.
+                // using X = int*[];
+                Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using X = int*[];").WithLocation(2, 1),
+                // (2,11): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                // using X = int*[];
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "int*").WithLocation(2, 11));
+        }
+
+        [Fact]
+        public void TestUnsafeAlias11()
+        {
+            var csharp = @"
+using unsafe X = int*[];
+
+class C
+{
+    void M(X t) { }
+}
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics(
+                // (6,12): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //     void M(X t) { }
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "X").WithLocation(6, 12));
+        }
+
+        [Fact]
+        public void TestUnsafeAlias12()
+        {
+            var csharp = @"
+using unsafe X = int*[];
+
+class C
+{
+    unsafe void M(X t) { }
+}
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestUnsafeAlias13()
+        {
+            var csharp = @"
+using unsafe X = int*[][];
+
+class C
+{
+    void M(X t) { }
+}
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics(
+                // (6,12): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //     void M(X t) { }
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "X").WithLocation(6, 12));
+        }
+
+        [Fact]
+        public void TestUnsafeAlias14()
+        {
+            var csharp = @"
+using unsafe X = int*[][];
+
+class C
+{
+    unsafe void M(X t) { }
+}
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void TestUnsafeAlias1_FP()
+        {
+            var csharp = @"
+using unsafe X = delegate*<int,int>;
+
+class C
+{
+    void M(X x) { }
+}
+";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics(
+                // (6,12): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
```

Comment:
Should the spec mention this check (unsafe alias can only be used in safe context)?

Reply:
I'll update the spec to be explicit.  However, the current unsafe-spec says thsi:

```c#
Other than establishing an unsafe context, thus permitting the use of pointer types, the unsafe modifier has no effect on a type or a member.
```

So it naturally falls out.  If you are using a pointer type (doesn't matter how you got it), you need to be in an unsafe context.